### PR TITLE
Prepare release 0.100.0

### PR DESCRIPTION
**Pull request by alejoe91**
- [x] Updated version in `pyproject.toml`
- [x] Comment out git installs in `pyproject.toml`
- [x] Add warning about frozen PRs in README
- [x] Add main release notes in `whatisnew.rst`
- [x] Add detailed changelog in `releases/0.100.0.rst`
- [x] Update doc version in README
- [x] Set `DEV_MODE=False` in `__init__.py`

### CODE REVIEW

File: doc/releases/0.100.0.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+Main changes:
+
+* Several improvements and bug fixes for Windows users
+* Important refactoring of NWB extractors:
+  * implemented direct backend implementation (to avoid using `pynwb`)
+  * sped up streaming using `remfile`
+  * added support for `zarr` backend
+* Removed `joblib` dependency in favor of `ParallelProcessExecutor`
+* Improved flexibility when running sorters in container by adding several options for installing `spikeinterface`
```
Comment (zm711) [inline doc/releases/0.100.0.rst:16]:
```suggestion
* Improved flexibility when running sorters in containers by adding several options for installing `spikeinterface`
```
or a sorter in a container.

File: doc/whatisnew.rst [hunk 2]
```diff
@@ -33,6 +34,19 @@ Release notes
...
+===============
+
+
+* Several improvements and bug fixes for Windows users
+* Important refactoring of NWB extractors:
+  * implemented direct backend implementation (to avoid using `pynwb`)
+  * sped up streaming using `remfile`
+  * added support for `zarr` backend
+* Removed `joblib` dependency in favor of `ParallelProcessExecutor`
+* Improved flexibility when running sorters in container by adding several options for installing `spikeinterface`
```
Comment (zm711) [inline doc/whatisnew.rst:47]:
```suggestion
* Improved flexibility when running sorters in containers by adding several options for installing `spikeinterface`
```

**Review by zm711 — COMMENTED** [review]
And switch documentation to point to 100 :)

**Comment by alejoe91** [comment]
Weird diff

**Pull request closed without merging**